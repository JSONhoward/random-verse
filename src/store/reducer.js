export const reducer = (state, action) => {
    let book, chapter, verse, toggle, displayVerse
    switch(action.type) {
        case 'LOADING_TRUE':
            state = {...state, loading: true}
            return state;
        case 'LOADING_FALSE':
            state = {...state, loading: false}
            return state;
        case 'SET_VERSE':
            displayVerse = action.payload
            state = {...state, displayVerse}
            return state;
        case 'TOGGLE_BOOK':
            toggle = !state.bookOptions
            state = {...state, bookOptions: toggle}
            return state;
        case 'TOGGLE_CHAPTER':
            toggle = !state.chapterOptions
            state = {...state, chapterOptions: toggle}
            return state;
        case 'TOGGLE_VERSE':
            toggle = !state.verseOptions
            state = {...state, verseOptions: toggle}
            return state;
        case 'CHOOSE_BOOK':
            book = action.payload
            chapter = '(Random)'
            verse = '(Random)'
            state = {...state, book, chapter, verse, bookOptions: false}
            return state;
        case 'CHOOSE_CHAPTER':
            chapter = action.payload
            state = {...state, chapter, chapterOptions: false}
            return state;
        case 'CHOOSE_VERSE':
            verse = action.payload
            state = {...state, verse, verseOptions: false}
            return state;
        case 'CLOSE_ALL':
                state = {
                    ...state,
                    bookOptions: state.bookOptions && false,
                    chapterOptions: state.chapterOptions && false,
                    verseOptions: state.verseOptions && false
                }
            return state;
        default:
            return state;
    }
}