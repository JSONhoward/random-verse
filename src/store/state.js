import {createContext} from 'react'

const homeContent = 'For God so loved the world, that he gave his only begotten Son, that whosoever believeth in him should not perish, but have everlasting life.'
const homeVerse = 'John 3:16'

export const initialState = {
    bookOptions: false,
    chapterOptions: false,
    verseOptions: false,
    book: '(Random)',
    chapter: '(Random)',
    verse: '(Random)',
    displayVerse: {
        verse: homeVerse,
        content: homeContent
    },
    loading: false
}

export const StateContext = createContext(null)