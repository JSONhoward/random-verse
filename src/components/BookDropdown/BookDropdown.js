import React, { useContext } from 'react'
import { FaCaretDown } from 'react-icons/fa'

import { StateContext } from '../../store/state'
import { BIBLE } from '../../utils/bible'
import {ListItem, DropdownDiv, Caret, OptionsDiv} from './styles'

const BookDropdown = () => {
    const [state, dispatch] = useContext(StateContext)
    const { bookOptions, book } = state

    let dropdownList = BIBLE.map((item, i) => {
        return (
            <ListItem
                tabIndex={1}
                id={i}
                onClick={() => chooseBook(item)}
                key={i}
                onKeyPress={e => {
                    if (e.charCode === 32 || e.charCode === 13) {
                        chooseBook(item)
                    }
                }}>
                {item.name}
            </ListItem>
        )
    })

    const chooseBook = option => {
        dispatch({ type: 'CHOOSE_BOOK', payload: option.name })
    }

    const keyOpen = (e) => {
        if (e.target.id === 'dropdown') {
            if (e.charCode === 13 || e.charCode === 32) {
                dispatch({ type: 'TOGGLE_BOOK' })
            }
        }
    }

    const clickOpen = () => {
        dispatch({ type: 'TOGGLE_BOOK' })
    }

    return (
        <DropdownDiv id={'dropdown'} onKeyPress={e => keyOpen(e)} tabIndex={1}>
            <p>{book}</p>
            <Caret onClick={clickOpen}>
                <FaCaretDown size={'1.5rem'} />
            </Caret>
            <OptionsDiv isOpen={bookOptions}>
                {dropdownList}
            </OptionsDiv>
        </DropdownDiv>
    )
}

export default BookDropdown
