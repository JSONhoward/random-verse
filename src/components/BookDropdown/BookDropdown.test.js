import React from 'react'
import { create } from 'react-test-renderer'
import BookDropdown from './BookDropdown'
import { initialState, StateContext } from '../../store/state'
import { reducer } from '../../store/reducer'

describe('BookDropdown Tests', () => {
    test('Snapshot Test', () => {
        const Component = () => {
            const [state, dispatch] = React.useReducer(reducer, initialState)
            return (
                <StateContext.Provider value={[state, dispatch]}>
                    <BookDropdown />
                </StateContext.Provider>
            )
        }
        const tree = create(<Component />).toJSON()

        expect(tree).toMatchSnapshot()
    })
})