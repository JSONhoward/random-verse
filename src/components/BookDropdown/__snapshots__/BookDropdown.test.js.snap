// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookDropdown Tests Snapshot Test 1`] = `
<div
  className="sc-AxjAm bbPthE"
  id="dropdown"
  onKeyPress={[Function]}
  tabIndex={1}
>
  <p>
    (Random)
  </p>
  <div
    className="sc-AxhCb OGtmh"
    onClick={[Function]}
  >
    <svg
      fill="currentColor"
      height="1.5rem"
      size="1.5rem"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 320 512"
      width="1.5rem"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
      />
    </svg>
  </div>
  <div
    className="sc-AxirZ cSkHFM"
  >
    <div
      className="sc-AxiKw lgMcRQ"
      id={0}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      (Random)
    </div>
    <div
      className="sc-AxiKw lgMcRQ"
      id={1}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      Genesis
    </div>
    <div
      className="sc-AxiKw lgMcRQ"
      id={2}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      Exodus
    </div>
    <div
      className="sc-AxiKw lgMcRQ"
      id={3}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      Leviticus
    </div>
    <div
      className="sc-AxiKw lgMcRQ"
      id={4}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      Numbers
    </div>
    <div
      className="sc-AxiKw lgMcRQ"
      id={5}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      Deuteronomy
    </div>
    <div
      className="sc-AxiKw lgMcRQ"
      id={6}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      Joshua
    </div>
    <div
      className="sc-AxiKw lgMcRQ"
      id={7}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      Judges
    </div>
    <div
      className="sc-AxiKw lgMcRQ"
      id={8}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      Ruth
    </div>
    <div
      className="sc-AxiKw lgMcRQ"
      id={9}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      1 Samuel
    </div>
    <div
      className="sc-AxiKw lgMcRQ"
      id={10}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      2 Samuel
    </div>
    <div
      className="sc-AxiKw lgMcRQ"
      id={11}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      1 Kings
    </div>
    <div
      className="sc-AxiKw lgMcRQ"
      id={12}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      2 Kings
    </div>
    <div
      className="sc-AxiKw lgMcRQ"
      id={13}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      1 Chronicles
    </div>
    <div
      className="sc-AxiKw lgMcRQ"
      id={14}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      2 Chronicles
    </div>
    <div
      className="sc-AxiKw lgMcRQ"
      id={15}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      Ezra
    </div>
    <div
      className="sc-AxiKw lgMcRQ"
      id={16}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      Nehemiah
    </div>
    <div
      className="sc-AxiKw lgMcRQ"
      id={17}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      Esther
    </div>
    <div
      className="sc-AxiKw lgMcRQ"
      id={18}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      Job
    </div>
    <div
      className="sc-AxiKw lgMcRQ"
      id={19}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      Psalms
    </div>
    <div
      className="sc-AxiKw lgMcRQ"
      id={20}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      Proverbs
    </div>
    <div
      className="sc-AxiKw lgMcRQ"
      id={21}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      Ecclesiastes
    </div>
    <div
      className="sc-AxiKw lgMcRQ"
      id={22}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      Song of Solomon
    </div>
    <div
      className="sc-AxiKw lgMcRQ"
      id={23}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      Isaiah
    </div>
    <div
      className="sc-AxiKw lgMcRQ"
      id={24}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      Jeremiah
    </div>
    <div
      className="sc-AxiKw lgMcRQ"
      id={25}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      Lamentations
    </div>
    <div
      className="sc-AxiKw lgMcRQ"
      id={26}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      Ezekiel
    </div>
    <div
      className="sc-AxiKw lgMcRQ"
      id={27}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      Daniel
    </div>
    <div
      className="sc-AxiKw lgMcRQ"
      id={28}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      Hosea
    </div>
    <div
      className="sc-AxiKw lgMcRQ"
      id={29}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      Joel
    </div>
    <div
      className="sc-AxiKw lgMcRQ"
      id={30}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      Amos
    </div>
    <div
      className="sc-AxiKw lgMcRQ"
      id={31}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      Obadiah
    </div>
    <div
      className="sc-AxiKw lgMcRQ"
      id={32}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      Jonah
    </div>
    <div
      className="sc-AxiKw lgMcRQ"
      id={33}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      Micah
    </div>
    <div
      className="sc-AxiKw lgMcRQ"
      id={34}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      Nahum
    </div>
    <div
      className="sc-AxiKw lgMcRQ"
      id={35}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      Habakkuk
    </div>
    <div
      className="sc-AxiKw lgMcRQ"
      id={36}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      Zephaniah
    </div>
    <div
      className="sc-AxiKw lgMcRQ"
      id={37}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      Haggai
    </div>
    <div
      className="sc-AxiKw lgMcRQ"
      id={38}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      Zechariah
    </div>
    <div
      className="sc-AxiKw lgMcRQ"
      id={39}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      Malachi
    </div>
    <div
      className="sc-AxiKw lgMcRQ"
      id={40}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      Matthew
    </div>
    <div
      className="sc-AxiKw lgMcRQ"
      id={41}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      Mark
    </div>
    <div
      className="sc-AxiKw lgMcRQ"
      id={42}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      Luke
    </div>
    <div
      className="sc-AxiKw lgMcRQ"
      id={43}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      John
    </div>
    <div
      className="sc-AxiKw lgMcRQ"
      id={44}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      Acts
    </div>
    <div
      className="sc-AxiKw lgMcRQ"
      id={45}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      Romans
    </div>
    <div
      className="sc-AxiKw lgMcRQ"
      id={46}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      1 Corinthians
    </div>
    <div
      className="sc-AxiKw lgMcRQ"
      id={47}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      2 Corinthians
    </div>
    <div
      className="sc-AxiKw lgMcRQ"
      id={48}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      Galatians
    </div>
    <div
      className="sc-AxiKw lgMcRQ"
      id={49}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      Ephesians
    </div>
    <div
      className="sc-AxiKw lgMcRQ"
      id={50}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      Philippians
    </div>
    <div
      className="sc-AxiKw lgMcRQ"
      id={51}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      Colossians
    </div>
    <div
      className="sc-AxiKw lgMcRQ"
      id={52}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      1 Thessalonians
    </div>
    <div
      className="sc-AxiKw lgMcRQ"
      id={53}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      2 Thessalonians
    </div>
    <div
      className="sc-AxiKw lgMcRQ"
      id={54}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      1 Timothy
    </div>
    <div
      className="sc-AxiKw lgMcRQ"
      id={55}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      2 Timothy
    </div>
    <div
      className="sc-AxiKw lgMcRQ"
      id={56}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      Titus
    </div>
    <div
      className="sc-AxiKw lgMcRQ"
      id={57}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      Philemon
    </div>
    <div
      className="sc-AxiKw lgMcRQ"
      id={58}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      Hebrews
    </div>
    <div
      className="sc-AxiKw lgMcRQ"
      id={59}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      James
    </div>
    <div
      className="sc-AxiKw lgMcRQ"
      id={60}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      1 Peter
    </div>
    <div
      className="sc-AxiKw lgMcRQ"
      id={61}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      2 Peter
    </div>
    <div
      className="sc-AxiKw lgMcRQ"
      id={62}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      1 John
    </div>
    <div
      className="sc-AxiKw lgMcRQ"
      id={63}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      2 John
    </div>
    <div
      className="sc-AxiKw lgMcRQ"
      id={64}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      3 John
    </div>
    <div
      className="sc-AxiKw lgMcRQ"
      id={65}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      Jude
    </div>
    <div
      className="sc-AxiKw lgMcRQ"
      id={66}
      onClick={[Function]}
      onKeyPress={[Function]}
      tabIndex={1}
    >
      Revelation
    </div>
  </div>
</div>
`;
